### ISSUE TITLE
Cannot stream camera video

### ISSUE THREAD

**Issue by jaykyll** [hjdhjd/homebridge-unifi-protect#737]
**Describe The Problem:**
After installing the plugin homebridge unifi protect, unable to stream video through Homekit. Got an error message.
My homebridge is installed on a mac mini.

**Logs:**
[5/16/2022, 11:11:52 PM] [UniFi Protect] JayUniFiCKG2P [UCK-G2-PLUS] JayUnifiG3I-Garage [UVC G3 Instant]: FFmpeg process ended unexpectedly with a signal received of SIGABRT.
[5/16/2022, 11:11:52 PM] [UniFi Protect] JayUniFiCKG2P [UCK-G2-PLUS] JayUnifiG3I-Garage [UVC G3 Instant]: FFmpeg command line that errored out was: ffmpeg -hide_banner -probesize 2048 -max_delay 500000 -r 30 -rtsp_transport tcp -i rtsps://10.0.0.2:7441/RU3JiBCsg2Wr5suz?enableSrtp -map 0:v -vcodec copy -payload_type 99 -ssrc 4775890 -f rtp -srtp_out_suite AES_CM_128_HMAC_SHA1_80 -srtp_out_params qvv+my+HGhRyqGdF+O4QHC20mjNxq7D2h/i5UDGs srtp://10.0.0.41:55518?rtcpport=55518&localrtcpport=55518&pkt_size=564
[5/16/2022, 11:11:52 PM] [UniFi Protect] dyld[28677]: Library not loaded: /usr/local/opt/libx11/lib/libX11.6.dylib

[5/16/2022, 11:11:52 PM] [UniFi Protect]   Referenced from: /usr/local/bin/ffmpeg

[5/16/2022, 11:11:52 PM] [UniFi Protect]   Reason: tried: '/usr/local/opt/libx11/lib/libX11.6.dylib' (no such file), '/usr/local/lib/libX11.6.dylib' (no such file), '/usr/lib/libX11.6.dylib' (no such file)

[5/16/2022, 11:11:52 PM] [UniFi Protect] JayUniFiCKG2P [UCK-G2-PLUS] JayUnifiG3I-Garage [UVC G3 Instant]: Stopped video streaming session.

```
Show the Homebridge logs here.
Remove any sensitive information.
```

**Homebridge Configuration:**

{
    "bridge": {
        "name": "Homebridge",
        "username": "",
        "port": 51826,
        "pin": ""
    },
    "description": "This is an example configuration file with one fake accessory and one fake platform. You can use this as a template for creating your own configuration file containing devices you actually own.",
    "accessories": [],
    "platforms": [
        {
            "name": "Config",
            "port": 8581,
            "platform": "config"
        },
        {
            "controllers": [
                {
                    "address": "X.X.X.X",
                    "password": "",
                    "username": ""
                }
            ],
            "platform": "UniFi Protect"
        }
    ]
}

**Screenshots:**

**Environment:**

* **Homebridge Version:** 1.4.1
* **Node Version:** 16.15.0
* **Homebridge-UniFi-Protect Plugin Version:** 5.5.0
* **UniFi Protect Version:** 1.21.5
* **UniFi Protect Controller Type:** UniFi Cloud Key Gen2 Plus
* **UniFi Protect Controller Firmware Version:** 2.3.15
* **Apple Device and iOS / macOS / iPadOS / tvOS Version:**
* **Operating System and OS Version:**  MacOS 12.1 on Mac mini (Late 2014)

**Comment by Stroh37** [comment]
I noticed you didn't post a unifi protect application version. If you're on EA firmware I know I was having problems streaming 100% of the time on protect 2.0 beta 5.

**Comment by jaykyll** [comment]
Sorry missed it.
It's updated on orignal post.

**Comment by hjdhjd** [comment]
Thanks for the report - you don’t have a working FFmpeg installation is the error it’s showing above. I would check to ensure that /usr/local/bin/ffmpeg works - it doesn’t appear so in the above.

Give you’re on a 2014 Mac…I’d also simply try uninstalling the plugin and reinstalling it. HBUP will incorporate a known working version of FFmpeg for Homebridge plugins by default. Sometimes this plugin doesn’t properly get installed the first time (it’s really an ffmpeg-for-homebridge issue).

In short:

1. Uninstall HBUP.
2. Reinstall HBUP.
3. Try to see if that works.
4. If not, install a working FFmpeg on your Mac and tell HBUP about it.

Steps 1+2 should be the clearest path to getting working.

**Comment by jaykyll** [comment]
Same result, what I cannot understand, even if I remove HBUP, in terminal window on my mac, the command "ffmpeg" give this result : 
dyld[34639]: Library not loaded: /usr/local/opt/libx11/lib/libX11.6.dylib
  Referenced from: /usr/local/bin/ffmpeg
  Reason: tried: '/usr/local/opt/libx11/lib/libX11.6.dylib' (no such file), '/usr/local/lib/libX11.6.dylib' (no such file), '/usr/lib/libX11.6.dylib' (no such file)
Abort trap: 6

Is it possible, there are other ffmpeg package installed? and if yes how to clean it completely?

**Comment by hjdhjd** [comment]
Sorry, that's beyond the scope of support I provide. Lots of references out there on installing FFmpeg-related packages for the Mac.

I'll offer two final pointers:

1. Try Homebrew - it's a package manager for the Mac and install FFmpeg that way. See: https://trac.ffmpeg.org/wiki/CompilationGuide/macOS#ffmpegthroughHomebrew
2. I would suggest trying the Discord - people there can point you in the right direction.

Your issue is that you don't have a working FFmpeg on your system...best of luck!

**Comment by Stroh37** [comment]
I have some experience with homebridge on Mac. I had to go into the plug-in settings and specify the location of ffmpeg. Look under your homebridge directory for an ffmpeg folder. Or install it via home brew. Once you have it found go deeper into the directory until you find ffmpeg executable. Then enter the executable ffmpeg complete location in the homebridge unifi protect plug-in settings. It'll be in the third drop down section.
